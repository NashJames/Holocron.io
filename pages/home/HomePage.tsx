import TitleSection from './TitleContainer/TitleSection'
import FetchSection from './FetchContainer/FetchSection'

export default function HomePage() {
  return (
    <>
      <TitleSection />
      <FetchSection />
    </>
  )
}
