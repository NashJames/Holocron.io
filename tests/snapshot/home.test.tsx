import { act, render } from '@testing-library/react'

import HomePage from '@pages/home'

test('renders home', async () => {
  const { container } = render(<HomePage />)
  await act(async () => expect(container).toMatchSnapshot())
})
