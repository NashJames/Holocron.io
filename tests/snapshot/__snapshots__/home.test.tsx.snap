// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home 1`] = `
<div>
  <section
    class="Content"
  >
    <div
      class="Title"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 className css-o2w69a-MuiTypography-root"
      >
        A modern API 
        <br />
        for the 
        <span
          class="className"
        >
          Jedi Archives
        </span>
      </h1>
    </div>
    <div
      class="Logo"
    >
      <div
        class="Wrapper"
      >
        <div
          class="Cube"
        >
          <img
            alt="Jedi Holocron SVG"
            class="Front"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <img
            alt="Jedi Holocron SVG"
            class="Back"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <img
            alt="Jedi Holocron SVG"
            class="Top"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <img
            alt="Jedi Holocron SVG"
            class="Bottom"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <img
            alt="Jedi Holocron SVG"
            class="Left"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <img
            alt="Jedi Holocron SVG"
            class="Right"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="Shadow"
        />
      </div>
    </div>
  </section>
  <section
    class="Content"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 Subtitle css-5lbw0b-MuiTypography-root"
    >
      Take a look at the data below...
    </h4>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd className css-l4u8b9-MuiInputBase-root-MuiInput-root"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <input
          aria-describedby="SearchBar-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-c63i49-MuiInputBase-input-MuiInput-input"
          id="SearchBar"
          placeholder="Search..."
          type="search"
          value=""
        />
        <span
          class="SearchButtons"
        >
          <button
            aria-label="Click here or press the 'Enter' key"
            class="MuiButtonBase-root MuiButton-root MuiButton-square MuiButton-squarePrimary MuiButton-sizeSmall MuiButton-squareSizeSmall MuiButton-root MuiButton-square MuiButton-squarePrimary MuiButton-sizeSmall MuiButton-squareSizeSmall css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="submit"
          >
            REQUEST
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Press the '/' key to begin searching"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall IconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <kbd
              aria-hidden="true"
            >
              /
            </kbd>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
        id="SearchBar-helper-text"
      />
    </div>
    <div
      class="cm-theme CodeMirror"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary LoadingCircle css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
      <div
        class="cm-editor ͼ1 ͼ3 ͼ4 ͼ1d ͼ15"
      >
        <div
          aria-live="polite"
          style="position: fixed; top: -10000px;"
        />
        <div
          class="cm-scroller"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="cm-gutters"
            style="min-height: 14px; position: sticky;"
          >
            <div
              class="cm-gutter cm-lineNumbers"
            >
              <div
                class="cm-gutterElement"
                style="height: 0px; visibility: hidden; pointer-events: none;"
              >
                9
              </div>
              <div
                class="cm-gutterElement"
                style="height: 14px;"
              >
                1
              </div>
            </div>
          </div>
          <div
            aria-autocomplete="list"
            aria-multiline="true"
            aria-readonly="true"
            autocapitalize="off"
            autocorrect="off"
            class="cm-content"
            contenteditable="true"
            data-language="json"
            role="textbox"
            spellcheck="false"
            style="tab-size: 4"
            translate="no"
          >
            <div
              class="cm-line"
            >
              <br />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="cm-layer cm-layer-above cm-cursorLayer"
            style="z-index: 150; animation-duration: 1200ms;"
          />
          <div
            aria-hidden="true"
            class="cm-layer cm-selectionLayer"
            style="z-index: -2;"
          />
        </div>
      </div>
    </div>
    <img
      alt="An illustration of the Death Star"
      class="DeathStar"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a red asteroid"
      class="Asteroid HighPosition Delay6s"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a red asteroid"
      class="Asteroid LowPosition Delay3s"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a blue asteroid"
      class="Asteroid MidPosition "
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a blue asteroid"
      class="Asteroid LowPosition Delay9s"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a pink asteroid"
      class="Asteroid HighPosition Delay5s"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="An illustration of a pink asteroid"
      class="Asteroid MidPosition Delay1s"
      data-nimg="1"
      decoding="async"
      draggable="false"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </section>
</div>
`;
